lib/capitano.js

```diff
@@ -0,0 +1,188 @@
+const _ = require('lodash')
+const Promise = require('bluebird')
+const preload = require('./preload')
+
+const LATEST = 'latest'
+
+function cmp (a, b) {
+  if (a < b) {
+    return -1
+  } else if (a === b) {
+    return 0
+  } else {
+    return 1
+  }
+}
+
+function selectApplication (resin, form, deviceType) {
+  // TODO: what if there is no matching app ?
+  return resin.models.application.getAll({filter: {device_type: deviceType}})
+  .then(function (applications) {
+    return form.ask({
+      message: 'Select an application',
+      type: 'list',
+      choices: applications.map(function (app) {
+        return {name: app.app_name, value: app}
+      })
+    })
+  })
+}
+
+function selectApplicationCommit (resin, form, appId) {
+  return resin.models.build.getAllByApplication(appId)
+  .then(function (builds) {
+    if (builds.length === 0) {
+      throw new Error('This application has no builds')
+    }
+    builds = builds.filter(function (build) {
+      return build.status === 'success'
+    }).sort(function (a, b) {
+      return -cmp(a.push_timestamp, b.push_timestamp)
+    })
+    const choices = builds.map(function (build) {
+      return {
+        name: build.push_timestamp + ' - ' + build.commit_hash,
+        value: build.commit_hash
+      }
+    })
+    choices.splice(0, 0, {'name': LATEST, 'value': LATEST})
+    return form.ask({
+      message: 'Select a build',
+      type: 'list',
+      default: LATEST,
+      choices: choices
+    })
+  })
+}
+
+function disableAutomaticUpdates (form, resin, application, commit) {
+  if ((commit === LATEST) || !application.should_track_latest_release) {
+    return Promise.resolve()
+  }
+  const message = (
+    'This application is set to automatically update all devices to the ' +
+    'latest available version.\n' +
+    'You need to disable this if you want to preload a specific build. ' +
+    'Do you want to disable automatic updates for this application?'
+  )
+  return form.ask({
+    message: message,
+    type: 'list',
+    default: true,
+    choices: [
+      {name: 'Yes', value: true},
+      {name: 'No', value: false}
+    ]
+  })
+  .then(function (update) {
+    if (!update) {
+      return
+    }
+    return resin.pine.patch({
+      resource: 'application',
+      id: application.id,
+      body: {
+        should_track_latest_release: false
+      }
+    })
+  })
+}
+
+module.exports = {
+  signature: 'preload <image>',
+  description: '(beta) preload an app on a disk image',
+  help: `Warning: 'resin preload' requires Docker to be correctly installed in
+your shell environment. For more information (including Windows support)
+please check the README here: https://github.com/resin-io/resin-cli .
+
+Use this command to preload an application to a local disk image.
+
+Examples:
+  $ resin preload resin.img --app 12345 --commit e1f2592fc6ee949e68756d4f4a48e49bff8d72a0 --splash-image some-image.png
+  $ resin preload resin.img`,
```

Comment:
So the standalone interface doesn't even use capitano? This is definitely something we should look at.